### DISCUSSION TITLE
Aborting and saving the final best model. Encountered exception:

### DISCUSSION THREAD

**Discussion by Parth-tech** [explosion/spaCy#8510] [Help: Coding & Implementations] [answered]
Aborting and saving the final best model. Encountered exception:

**_RuntimeError('CUDA out of memory. Tried to allocate 1.01 GiB (GPU 0; 14.76 GiB
total capacity; 11.96 GiB already allocated; 1017.75 MiB free; 12.84 GiB
reserved in total by PyTorch)')_**

I am current training sciBert on my dataset which I have split into numerous .spacy files. My original data comprises of 65 million entries. (IOB Format)

I saw that plenty of them have encountered this issue, but despite changing the variable like:
**_1. batch_size ( training )
2. max_batch_items ( transformer )
3. size ( training batcher )_**
 This issue still persists. 

Could you help with how to deal and what precisely causes this?

Kindly expound the answer, I am a novice in this domain. 

Thank you so much for your assistance.
[base_config_new (1).txt](https://github.com/explosion/spaCy/files/6720094/base_config_new.1.txt)
!

**Comment by polm** [comment, marked as answer]
Like I mentioned in #8506, you should look at using the streaming corpus reader. To start you can set `max_epochs = -1`. Depending on your data you might need to do some other things like initialize the labels. [The docs](https://spacy.io/usage/training#custom-code-readers-batchers) explain how to do that.

What kind of model are you training?

If you're new to all this, it may be easier to start with a smaller portion of your dataset and train a model on that before scaling up to the whole thing.

**Reply by Parth-tech** [reply]
Oh Thank You @polm. I am training Sci_Bert Model for NER on Covid-19 dataset. The dataset was huge so I created numerous Chunks of it as you had guided. And post that I set epoch = -1 too yet there were issue faced. Yet, I'll give it a try post reading the mentioned references.

Thanks alot!
